lib/robots/sdr_repo/preservation_ingest/complete_ingest.rb

```diff
@@ -28,14 +28,27 @@ def complete_ingest
         end
 
         def remove_deposit_bag
-          deposit_bag_pathname.rmtree
+          rm_deposit_bag_safely_for_ceph
         rescue StandardError => e
           errmsg = "Error completing ingest for #{druid}: failed to remove deposit bag (#{deposit_bag_pathname}): " \
             "#{e.message}\n + e.backtrace.join('\n')"
           LyberCore::Log.error(errmsg)
           raise(ItemError, errmsg)
         end
 
+        # we stat all the files in the Moab in hopes of preventing issues when reading metadata about
+        # files in the newly created Moab version.  this addresses what we suspect to be interplay between
+        # Ceph backed storage and our use of hardlinking (instead of e.g. copying) to get content from the
+        # deposit bag to the new Moab version.  see https://github.com/sul-dlss/preservation_catalog/issues/1633
+        def rm_deposit_bag_safely_for_ceph
+          deposit_bag_pathname.rmtree
+          stat_moab_dir_contents
```

Comment:
wouldn't you want to do these in the opposite order?

Reply:
ah, no, that ordering was intentional (and figuring out how to test the order, since it's important in this case, was why writing the test took me longer than i wanted -- turns out there's a handy rspec built-in for the rare case where you want to test a specific call order).

short version of the explanation in https://github.com/sul-dlss/preservation_catalog/issues/1633 --
* "The only issue we've seen is if you hardlink b to a, then rm a, then never stat b, the inode is added to the "stray" directory. By default there is a limit of 1 million stray entries -- so if you accumulate files in this state eventually users will be unable to rm any files, until you stat the b files."  -- some rando from a mailing list
* so our thinking was that the deposit dir is a, the moab contents are b, and b is acting wonky because we `rm` the original link to the file content, without ever `stat`ing the newer hardlinked content.

so, the idea is "`rm` a, then `stat` b".

we're not actually sure this'll resolve the issue, but it seems harmless enough if it doesn't.

and actually, a quick glance at some other moab versioning code yesterday made me think the new moab version was validated between when it gets created and when the deposit bag gets removed.  thus, it seems there's already a walk of the moab before the `rmtree` on the deposit bag.